.github/scripts/platform_lib.py

```diff
@@ -0,0 +1,377 @@
+import abc
+
+import sys
+import boto3
+import os
+from enum import Enum
+from fabric.api import *
+from ci_variables import local_fsim_dir, ci_azure_sub_id
+import datetime
+import pytz
+
+from azure.mgmt.resource import ResourceManagementClient
+from azure.identity import DefaultAzureCredential
+from azure.mgmt.compute import ComputeManagementClient
+import azure.mgmt.resourcegraph as arg
+
+from typing import Any, Callable, Dict, List
+
+# Reuse manager utilities
+# Note: ci_firesim_dir must not be used here because the persistent clone my not be initialized yet.
+sys.path.append(local_fsim_dir + "/deploy")
+from awstools.awstools import get_instances_with_filter
+
+# This tag is common to all instances launched as part of a given workflow
+unique_tag_key = 'ci_workflow_id'
```

Comment:
I've always not liked the naming of this var. Can we rename it to something like "workflow_id_tag"?